src/server/scripts/Northrend/Naxxramas/boss_four_horsemen.cpp

```diff
@@ -690,70 +690,77 @@ class boss_four_horsemen_sir : public CreatureScript
+                     break;
+                 case 3:
+                     damage = 1000;
+                     break;
+                 case 4:
+                     damage = 1500;
+                     break;
+                 case 5:
+                     damage = 4000;
+                     break;
+                 case 6:
+                     damage = 12000;
+                     break;
+                 default:
+                     damage = 20000 + 1000 * (GetStackAmount() - 7);
+                     break;
+             }
+             if (damage)
+             {
+                 CastSpellExtraArgs args(TRIGGERED_FULL_MASK);
+                 args.AddSpellBP0(damage);
+                 caster->CastSpell(GetTarget(), SPELL_MARK_DAMAGE, args);
+             }
+         }
+     }
+
+     void Register() override
+     {
+         AfterEffectApply += AuraEffectApplyFn(spell_four_horsemen_mark::OnApply, EFFECT_0, SPELL_AURA_DUMMY, AURA_EFFECT_HANDLE_REAL_OR_REAPPLY_MASK);
+     }
+};
 
-            void OnApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
-            {
-                if (Unit* caster = GetCaster())
-                {
-                    int32 damage;
-                    switch (GetStackAmount())
-                    {
-                        case 1:
-                            damage = 0;
-                            break;
-                        case 2:
-                            damage = 500;
-                            break;
-                        case 3:
-                            damage = 1000;
-                            break;
-                        case 4:
-                            damage = 1500;
-                            break;
-                        case 5:
-                            damage = 4000;
-                            break;
-                        case 6:
-                            damage = 12000;
-                            break;
-                        default:
-                            damage = 20000 + 1000 * (GetStackAmount() - 7);
-                            break;
-                    }
-                    if (damage)
-                    {
-                        CastSpellExtraArgs args(TRIGGERED_FULL_MASK);
-                        args.AddSpellBP0(damage);
-                        caster->CastSpell(GetTarget(), SPELL_MARK_DAMAGE, args);
-                    }
-                }
-            }
+ class spell_four_horsemen_consumption : public SpellScript
+ {
+     PrepareSpellScript(spell_four_horsemen_consumption);
 
-            void Register() override
-            {
-                AfterEffectApply += AuraEffectApplyFn(spell_four_horsemen_mark_AuraScript::OnApply, EFFECT_0, SPELL_AURA_DUMMY, AURA_EFFECT_HANDLE_REAL_OR_REAPPLY_MASK);
-            }
-        };
+     void HandleDamageCalc(SpellEffIndex /*effIndex*/)
+     {
+         SetHitDamage(2750);
+         if (GetCaster()->GetMap()->IsHeroic())
```

Comment:
can crash here?

Reply:
Doubt this.

Reply:
nope. caster is never null in SpellScript and nothing can cast a spell without being on map